numpy/_core/src/multiarray/item_selection.c

```diff
@@ -1137,7 +1116,10 @@ PyArray_Choose(PyArrayObject *ip, PyObject *op, PyArrayObject *out,
     PyDataMem_FREE(mps);
     if (out != NULL && out != obj) {
         Py_INCREF(out);
-        PyArray_ResolveWritebackIfCopy(obj);
+        if (PyArray_CopyAnyInto(out, obj) != 0) {
```

Comment:
```suggestion
        if (PyArray_CopyAnyInto(out, obj) < 0) {
```

Slightly more defensive in case another non-error return code is added in the future (almost certainly won't happen but a good defensive programming habit to get into).

Reply:
That makes sense; I made this change, thanks.